### PR TITLE
kernel: package x86-optimized crypto-misc modules

### PR DESCRIPTION
**Pull request by danielg4**
Some of the modules in the crypto-misc package have alternate
implementations optimized for different x86 instruction set extensions,
but only one of these was built for this package until now: twofish-i586.ko

Tested only with insmod, on both x86 and x86_64. The modules still do not
have an autoload, so the included dependencies must be manually loaded in
the correct order, which is not shown here.

Signed-off-by: Daniel Gimpelevich <daniel@gimpelevich.san-francisco.ca.us>

### CODE REVIEW

**Comment by hauke** [comment]
Why not add a autoprobe section for them?

**Comment by danielg4** [comment]
I could do so if that is desired. I assumed it was not previously done because it was undesirable.

File: package/kernel/linux/modules/crypto.mk [hunk 1]
```diff
@@ -646,9 +659,33 @@ endef
...
   define KernelPackage/crypto-misc/x86
     FILES+=$(LINUX_DIR)/arch/x86/crypto/twofish-i586.ko
+    FILES+=$(LINUX_DIR)/arch/x86/crypto/serpent-sse2-i586.ko
+    FILES+=$(LINUX_DIR)/arch/x86/crypto/glue_helper.ko
+    FILES+=$(LINUX_DIR)/crypto/ablk_helper.ko
+    FILES+=$(LINUX_DIR)/crypto/cryptd.ko
+    FILES+=$(LINUX_DIR)/crypto/gf128mul.ko
+    FILES+=$(LINUX_DIR)/crypto/lrw.ko
+    FILES+=$(LINUX_DIR)/crypto/xts.ko
   endef
```
Comment (blogic) [inline package/kernel/linux/modules/crypto.mk]: why not use multi line with \

Comment (danielg4) [reply package/kernel/linux/modules/crypto.mk]: I figured there just weren't very many there, so it wasn't worth rewriting the existing line. The difference is cosmetic anyway.

**Review by pprindeville — APPROVED** [review]
LGTM

**Comment by pprindeville** [comment]
> Why not add a autoprobe section for them?

I agree.  This would be better.

**Comment by danielg4** [comment]
Is this better?

File: package/kernel/linux/modules/crypto.mk [hunk 2]
```diff
@@ -655,7 +748,6 @@ endef
 
 $(eval $(call KernelPackage,crypto-test))
 
-
```
Comment (pprindeville) [inline package/kernel/linux/modules/crypto.mk]: Please avoid unnecessary whitespace changes.

File: package/kernel/linux/modules/crypto.mk [hunk 3]
```diff
@@ -645,7 +645,6 @@ endef
 
 $(eval $(call KernelPackage,crypto-sha512))
 
-
```
Comment (pprindeville) [inline package/kernel/linux/modules/crypto.mk]: Please undo this

File: package/kernel/linux/modules/crypto.mk [hunk 4]
```diff
@@ -645,6 +645,7 @@ endef
 
 $(eval $(call KernelPackage,crypto-sha512))
 
+
```
Comment (pprindeville) [inline package/kernel/linux/modules/crypto.mk]: Again, please avoid unnecessary whitespace changes.

File: package/kernel/linux/modules/crypto.mk (hunk 2 above)
Comment (danielg4) [reply package/kernel/linux/modules/crypto.mk]: These were accidental through Github's web interface, sorry. I know there is a way to squash the commits into the first one.

**Review by pprindeville — CHANGES_REQUESTED** [review]
Looks better.  Can you squash it now, please?

**Comment by danielg4** [comment]
Done. I see no reason not to merge this into the 18.06 branch also.

**Review by pprindeville — COMMENTED** [review]
Still some pending questions...

File: package/kernel/linux/modules/crypto.mk [hunk 5]
```diff
@@ -441,8 +441,21 @@ $(eval $(call KernelPackage,crypto-michael-mic))
...
+	CONFIG_CRYPTO_TWOFISH_X86_64_3WAY \
+	CONFIG_CRYPTO_SERPENT_SSE2_X86_64 \
+	CONFIG_CRYPTO_CAMELLIA_AESNI_AVX_X86_64 \
+	CONFIG_CRYPTO_CAST5_AVX_X86_64 \
+	CONFIG_CRYPTO_CAST6_AVX_X86_64 \
+	CONFIG_CRYPTO_TWOFISH_AVX_X86_64 \
+	CONFIG_CRYPTO_SERPENT_AVX_X86_64 \
+	CONFIG_CRYPTO_CAMELLIA_AESNI_AVX2_X86_64 \
+	CONFIG_CRYPTO_SERPENT_AVX2_X86_64 \
+	CONFIG_CRYPTO_SERPENT_SSE2_586 \
```
[the remark this answers is not in the document]
Comment (danielg4) [reply package/kernel/linux/modules/crypto.mk]: The build system filters out Kconfig symbols that do not exist, and these will exist only as needed.

**Review by pprindeville — COMMENTED** [review]
No further issues then.
